### ISSUE TITLE
Window.open crash in Sandbox mode

### ISSUE THREAD

**Issue by AshDevFr** [electron/electron#9422]
* Electron version: 1.6.8
* Operating system: MacOS 10.12.4 

### Expected behavior

When opening a new Window using `window.open` it should work well

### Actual behavior

We are using `sandbox mode` to use `window.open` natively.

When the application is loaded, we open 2 hidden new Windows with `window.open`. 

When the application launch we have this error :
```js
Uncaught Error: Unable to send AtomViewHostMsg_Message_Sync
    at EventEmitter.ipcRenderer.sendSync (<anonymous>:282:29)
    at Object.exports.require (<anonymous>:626:34)
    at Object.require.fs.electron (<anonymous>:5092:45)
    at s (<anonymous>:2:262)
    at <anonymous>:2:313
    at Object.<anonymous> (<anonymous>:737:12)
    at Object.require.10.../common/atom-binding-setup (<anonymous>:794:4)
    at s (<anonymous>:2:262)
    at e (<anonymous>:2:433)
    at <anonymous>:2:451
```

Here is the MacOS Error report we have when the app crash: https://gist.github.com/AshDevFr/8b5df15d2c20029d8757e7ff27dcd2bd

If we don't open any window with `window.open`, so the application don't crash

We use this code to handle window.open like indicated in the doc (https://github.com/electron/electron/blob/f64d82cf16b463a3b6e7d83172317c0545623cc9/docs/api/web-contents.md):

```js
this._window.webContents.on('new-window', (event, url, frameName, disposition, options) => {
  event.preventDefault();
  const display = disposition === 'new-window';
  event.newGuest = new BrowserWindow(Object.assign(options, {
    minWidth: 500,
    minHeight: 400,
    show: display
  }));
});
```

It seems to be link somehow to this instruction `api_web_contents->DestroyWebContents(true /* async */);` in https://github.com/electron/electron/blob/master/atom/browser/api/atom_api_web_contents.cc#L493

If we remove this line and rebuild electron it is working. (We currently use a fork of electron to remove this line and make it work)

### How to reproduce

I don't really know how to reproduce it because our App is big but I'll try to create a Repo with the same behavior.

**Comment by tarruda** [comment]
@AshDevFr Can you reproduce this before 1.6.8?

**Comment by AshDevFr** [comment]
@tarruda Yes, we updated to 1.6.8 lately and we were in 1.6.2 before (with a fork to remove the issue).
The issue was already existing before and appeared when this line appeared, this is how we discovered that this line was the issue for us. This is the commit https://github.com/electron/electron/commit/8f63fdb278ccefa4847748637ec69ee04c887ba7

Unfortunately we have leak when guest windows close, so we figured that maybe make it work with that line could help.

**Comment by tarruda** [comment]
> Unfortunately we have leak when guest windows close, so we figured that maybe make it work with that line could help

The leak should be fixed but another bug was introduced: https://github.com/electron/electron/pull/9389 . I was hoping it would make into 1.6.9 but didn't happen so you better wait use 1.6.7 for now.

**Comment by AshDevFr** [comment]
Great I hope the leak fix will come soon. 

About the other issue, do you have any idea of what could cause it ?

**Comment by tarruda** [comment]
> About the other issue, do you have any idea of what could cause it ?

It was already there. Basically it happened because the rpc server only stored 1 reference per remote object per webcontents, but it is possible that the same webcontents has two references to the same object since when a sandbox window is reloaded a new context is created before the old one is freed.

Because of the leak before #9314, old window contexts after by reload were never freed, so the bug was hidden.

**Comment by tarruda** [comment]
BTW, the leak fix is already merged, but since it ended up exposing #9389 I don't recommend upgrading yet.
